conans/client/cmake.py

```diff
@@ -74,6 +75,35 @@ def _generator(self):
 
         raise ConanException("Unknown cmake generator for these settings")
 
+    def _cmake_compiler_options(self, target_os, target_arch, target_comp):
+        host_platform = platform.system()
+        same_platform = (target_os == host_platform)
+
+        cmake_flags = []
+
+        if not same_platform:
+            cmake_flags.append("-DCMAKE_SYSTEM_NAME=%s" % target_os)
```

Comment:
CMAKE_SYSTEM_NAME should be "Darwin" if Macos is the target_os, and "Generic" if target_os is unknown/None.